src/text/raw_text_writer.rs

```diff
@@ -59,43 +68,44 @@ impl RawTextWriterBuilder {
     /// ```
     pub fn pretty() -> RawTextWriterBuilder {
         RawTextWriterBuilder {
-            // Each value appears on its own line
-            space_between_values: String::from("\n"),
-            // Values get four spaces of indentation per level of depth
-            indentation: String::from("    "),
-            // Field names and values are separated by a single space
-            space_after_field_name: String::from(" "),
-            // The first value in a container appears on a line by itself
-            space_after_container_start: String::from("\n"),
+            whitespace_config: WhitespaceConfig { ..PRETTY_WS_CONFIG },
         }
     }
 
-    pub fn with_space_between_values<S: Into<String>>(
+    pub fn with_space_between_top_values(
+        mut self,
+        space_between_top_values: &'static str,
+    ) -> RawTextWriterBuilder {
+        self.whitespace_config.space_between_top_level_values = space_between_top_values;
+        self
+    }
+
+    pub fn with_space_between_values(
         mut self,
-        space_between_values: S,
+        space_between_values: &'static str,
     ) -> RawTextWriterBuilder {
-        self.space_between_values = space_between_values.into();
+        self.whitespace_config.space_between_nested_values = space_between_values;
         self
     }
 
-    pub fn with_indentation<S: Into<String>>(mut self, indentation: S) -> RawTextWriterBuilder {
-        self.indentation = indentation.into();
+    pub fn with_indentation(mut self, indentation: &'static str) -> RawTextWriterBuilder {
+        self.whitespace_config.indentation = indentation;
         self
     }
 
-    pub fn with_space_after_field_name<S: Into<String>>(
+    pub fn with_space_after_field_name(
         mut self,
-        space_after_field_name: S,
+        space_after_field_name: &'static str,
     ) -> RawTextWriterBuilder {
-        self.space_after_field_name = space_after_field_name.into();
+        self.whitespace_config.space_after_field_name = space_after_field_name;
         self
     }
 
-    pub fn with_space_after_container_start<S: Into<String>>(
+    pub fn with_space_after_container_start(
         mut self,
-        space_after_container_start: S,
+        space_after_container_start: &'static str,
     ) -> RawTextWriterBuilder {
-        self.space_after_field_name = space_after_container_start.into();
+        self.whitespace_config.space_after_field_name = space_after_container_start;
```

Comment:
There's some dissonance here that I probably caused but can't remember why. Should `space_after_container_start` and `space_after_field_name` have two separate setters? Or should we unify their names somehow?

Reply:
I think this is just a bug? Here's the relevant snippet of the compact text format configuration: 

```
    // Single space between field names and values
    space_after_field_name: " ",
    // The first value in a container appears next to the opening delimiter
    space_after_container_start: "",
```

and here's the relevant snippet of the pretty text format configuration:

```
    // Field names and values are separated by a single space
    space_after_field_name: " ",
    // The first value in a container appears on a line by itself
    space_after_container_start: "\n",
```

Both of them have different values for these two fields, and those different values will be used. It's only this setter which conflates them, and no tests exercise this setter.

Reply:
I added some tests that exercise the builder setters.